xbmc/addons/GUIDialogAddonInfo.cpp

```diff
@@ -460,6 +488,69 @@ void CGUIDialogAddonInfo::OnSettings()
   CGUIDialogAddonSettings::ShowForAddon(m_localAddon);
 }
 
+void CGUIDialogAddonInfo::ShowDependencyList(const ADDONDEPS& deps)
+{
+  auto pDialog = g_windowManager.GetWindow<CGUIDialogSelect>(WINDOW_DIALOG_SELECT);
+  CFileItemList items;
+  for (auto& it : deps)
+  {
+    AddonPtr dep_addon, local_addon;
+    CServiceBroker::GetAddonMgr().FindInstallableById(it.first, dep_addon);
+    CServiceBroker::GetAddonMgr().GetAddon(it.first, local_addon);
+    if (dep_addon)
+    {
+      CFileItemPtr item(new CFileItem(dep_addon->Name()));
+      std::stringstream str;
+      str << it.first << " " << it.second.first.asString();
+      if (!it.second.second || !local_addon)
+        str << " " << StringUtils::Format(g_localizeStrings.Get(39022).c_str(),
+                                          local_addon ? g_localizeStrings.Get(39019).c_str()
+                                                      : g_localizeStrings.Get(39018).c_str());
+      else if (it.second.second && local_addon)
+        str << " " << StringUtils::Format(g_localizeStrings.Get(39023).c_str(),
+                                          g_localizeStrings.Get(39019).c_str(),
+                                          g_localizeStrings.Get(39018).c_str());
+
+      item->SetLabel2(str.str());
+      item->SetIconImage(dep_addon->Icon());
+      item->SetProperty("addon_id", it.first);
+      items.Add(item);
+    }
+    else
+    {
+      CFileItemPtr item(new CFileItem(it.first));
+      item->SetLabel2(g_localizeStrings.Get(161));
+      items.Add(item);
+    }
+  }
+
+  CFileItemPtr backup_item = GetCurrentListItem();
+  std::string result="1";
+  while (!result.empty())
+  {
+    result.clear();
+    pDialog->Reset();
+    pDialog->SetHeading(39017);
+    pDialog->SetUseDetails(true);
+    for (auto& it : items)
+      pDialog->Add(*it);
+    pDialog->Open();
+    if (pDialog->IsConfirmed())
+    {
+      const CFileItemPtr& item = pDialog->GetSelectedFileItem();
+      result = item->GetProperty("addon_id").asString();
+      AddonPtr dep_addon;
```

Comment:
Should we do the following to reduce the scope for `result` and save some calls.

```
while(true)
{
  if (result.empty())
    break;
}
```